app/core/items.ts

```diff
@@ -0,0 +1,95 @@
+import { EffectClass } from "../types/enum/EffectClass"
+import Board from "./board"
+import { PokemonEntity } from "./pokemon-entity"
+import { Item } from "../types/enum/Item"
+import PokemonState from "./pokemon-state"
+import { AbilityStrategies } from "./abilities/abilities"
+import { Ability } from "../types/enum/Ability"
+
+export function triggerItemEffects(
+    effect_class: EffectClass,
+    params: {
+        pokemon: PokemonEntity,
+        state: PokemonState,
+        board: Board,
+        target: PokemonEntity,
+        crit: boolean
+}){
+    params.pokemon.items.forEach((i) =>{
+        const handler = itemMap.get(i)
+        if(handler){
+            handler(effect_class, params)
+        } else{
+            console.error(`${i} missing from itemMap`)

```

Comment:
There is a logger for this.

Reply:
Yup was just me checking it. File didn't update properly